deepspeed/runtime/engine.py

```diff
@@ -114,14 +114,11 @@
 
 def split_half_float_double_sparse(tensors):
     device_type = get_accelerator().device_name()
-    supported_types = [
-        "torch.{}.HalfTensor".format(device_type), "torch.{}.FloatTensor".format(device_type),
-        "torch.{}.DoubleTensor".format(device_type), "torch.{}.BFloat16Tensor".format(device_type),
-        SparseTensor.type()
-    ]
-
+    supported_types = get_accelerator().supported_dtypes()
+    supported_types.append(SparseTensor.type())
```

Comment:
@tjruwase and I just discussed offline.

Let's remove this line (supported_types.append(SparseTensor.type())), then also add an attribute in SparseTensor to stash the dense tensor dtype like this:

`self.dtype = self.orig_dense_tensor`

Inside here: 
https://github.com/microsoft/DeepSpeed/blob/0a0819b78554a5130b7c1b31d18779d39421dade/deepspeed/runtime/sparse_tensor.py#L16